src/AppInstallerCommonCore/Versions.cpp

```diff
@@ -275,4 +275,35 @@ namespace AppInstaller::Utility
 
         return m_version < other.m_version;
     }
+
+    FourPartsVersionNumber::FourPartsVersionNumber(UINT64 version)
+    {
+        UINT64 mask16 = (1 << 16) - 1;
+        UINT64 revision = version & mask16;
+        UINT64 build = (version >> 0x10) & mask16;
+        UINT64 minor = (version >> 0x20) & mask16;
+        UINT64 major = (version >> 0x30) & mask16;
+
+        std::stringstream ssVersion;
+        ssVersion << major
+            << Version::DefaultSplitChars << minor
+            << Version::DefaultSplitChars << build
+            << Version::DefaultSplitChars << revision;
+        Assign(ssVersion.str(), Version::DefaultSplitChars);

```

Comment:
Not 100% sure how the compiler approaches this, but my guess is that we don't need to explicitly move the data here (in this case it's already an `rvalue` from the function call) and leave it for the compiler to optimize. Let me know if you have any suggestions or if you have some tests/experiments in mind you want me to execute.

Reply:
Looks like it isn't available until C++20:
https://en.cppreference.com/w/cpp/io/basic_stringstream/str

<html>
<body>
<!--StartFragment-->

std::basic_string<CharT,Traits,Allocator> str() &&; | (3) | (since C++20)
-- | -- | --


<!--EndFragment-->
</body>
</html>